cli/oauth.go

```diff
@@ -1 +1,120 @@
 package cli
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/coreos/issue-sync/cfg"
+	"github.com/dghubble/oauth1"
+)
+
+// getJIRAHTTPClient obtains an access token (either from configuration
+// or from an OAuth handshake) and creates an HTTP client that uses the
+// token, which can be used to configure a JIRA client.
+func getJIRAHTTPClient(config cfg.Config) (*http.Client, error) {
+	ctx := context.Background()
+
+	oauthConfig, err := getOauthConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	tok, ok := jiraTokenFromConfig(config)
+	if !ok {
+		tok, err = jiraTokenFromWeb(oauthConfig)
+		if err != nil {
+			return nil, err
+		}
+		config.SetJIRAToken(tok)
+	}
+
+	return oauthConfig.Client(ctx, tok), nil
+}
+
+// getOauthConfig parses a private key and consumer key from the
+// configuration, and creates an OAuth configuration which can
+// be used to begin a handshake.
+func getOauthConfig(config cfg.Config) (oauth1.Config, error) {
```

Comment:
Same here, this should really just be called `func oauthConfig(...)` or `func newOauthConfig(...)`